ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -253,7 +253,14 @@ class LocalTrainingService implements TrainingService {
 
             return Promise.resolve();
         }
-        tkill(trialJob.pid, 'SIGKILL');
+        tkill(trialJob.pid, 'SIGTERM');
+        const pid = trialJob.pid;
+        setTimeout((pid: number) => {
+            if (isAlive(pid)) {
+                tkill(pid, 'SIGKILL');
+            }
+        }, 5 * 1000, pid);
```

Comment:
with this solution, we have to wait for 5 seconds even the trail is stopped within 1 second?

Reply:
no, `setTimeout()` will not block the function, just put callback into the queue after delay.